projects/js-packages/charts/src/components/line-chart/line-chart.tsx

```diff
@@ -142,10 +143,18 @@ const LineChart: FC< LineChartProps > = ( {
 
 	const dataSorted = useMemo(
 		() =>
-			data.map( series => ( {
-				...series,
-				data: series.data.sort( ( a, b ) => a.date.getTime() - b.date.getTime() ),
-			} ) ),
+			data
+				.map( series => ( {
+					...series,
+					data: series.data.map( point => ( {
+						...point,
+						date: point.date ? point.date : parseAsLocalDate( point.dateString ),
+					} ) ),
+				} ) )
+				.map( series => ( {
+					...series,
+					data: series.data.sort( ( a, b ) => a.date.getTime() - b.date.getTime() ),
+				} ) ),
```

Comment:
I think this is functionally the same, more efficient, and possibly easier to read?

```suggestion
				.map( series => ( {
					...series,
					data: series.data
						.map( point => ( {
							...point,
							date: point.date ? point.date : parseAsLocalDate( point.dateString ),
						} ) )
						.sort( ( a, b ) => a.date.getTime() - b.date.getTime() ),
				} ) )
```

Reply:
Good point!